qos-service-impl/src/test/java/com/palantir/atlasdb/qos/ratelimit/CassandraMetricMeasurementsLoaderTest.java

```diff
@@ -0,0 +1,129 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import org.jmock.lib.concurrent.DeterministicScheduler;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.qos.config.CassandraHealthMetric;
+import com.palantir.atlasdb.qos.config.CassandraHealthMetricMeasurement;
+import com.palantir.atlasdb.qos.config.ImmutableCassandraHealthMetric;
+import com.palantir.atlasdb.qos.config.ImmutableCassandraHealthMetricMeasurement;
+import com.palantir.cassandra.sidecar.metrics.CassandraMetricsService;
+
+public class CassandraMetricMeasurementsLoaderTest {
+    private static final double LOWER_LIMIT = 0.0;
+    private static final double UPPER_LIMIT = 10.0;
+    private static final double METRIC_VALUE_1 = 15.0;
+    private static final double METRIC_VALUE_2 = 20.0;
+    private static final String METRIC_TYPE_1 = "metricType1";
+    private static final String METRIC_TYPE_2 = "metricType2";
+    private static final String METRIC_NAME_1 = "metricName1";
+    private static final String METRIC_NAME_2 = "metricName2";
+    private static final String METRIC_ATTRIBUTE_1 = "metricAttribute1";
+    private static final String METRIC_ATTRIBUTE_2 = "metricAttribute2";
+
+    private DeterministicScheduler scheduledExecutorService;
+    private Supplier<List<CassandraHealthMetric>> healthMetricSupplier;
+    private CassandraMetricsService cassandraMetricClient;
+    private CassandraMetricMeasurementsLoader cassandraMetricMeasurementsLoader;
+
+    @Before
+    public void setup() {
+        healthMetricSupplier = mock(Supplier.class);
+        cassandraMetricClient = mock(CassandraMetricsService.class);
+        scheduledExecutorService = new DeterministicScheduler();
+
+        cassandraMetricMeasurementsLoader = new CassandraMetricMeasurementsLoader(
+                healthMetricSupplier, cassandraMetricClient, scheduledExecutorService);
+    }
+
+    @Test
+    public void metricsLoaderLoadsMetricsEveryFiveSeconds() throws Exception {
+        when(healthMetricSupplier.get()).thenReturn(ImmutableList.of());
+
+        scheduledExecutorService.tick(1, TimeUnit.SECONDS);
+        verify(healthMetricSupplier, times(1)).get();
+
+        scheduledExecutorService.tick(4, TimeUnit.SECONDS);
+        verify(healthMetricSupplier, times(2)).get();
+
+        scheduledExecutorService.tick(5, TimeUnit.SECONDS);
+        verify(healthMetricSupplier, times(3)).get();
+
+        scheduledExecutorService.tick(5, TimeUnit.SECONDS);
+        verify(healthMetricSupplier, times(4)).get();
+    }
+
+    @Test
+    public void canReturnTheCassandraMetricsFromTheCassandraMetricService() throws Exception {
```

Comment:
Might be worth verifying the behavior if a call for cassandra metrics fails or the metric doesn't exist

Reply:
Done!